test/unit-test/CMakeLists.txt

```diff
@@ -1,92 +1,87 @@
-        ON )
-
-# Set output directories.
-set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
-set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
-set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
-
-# ====================== coreHTTP library configurations =======================
-
 # Include filepaths for source and include.
 include( ${MODULE_ROOT_DIR}/httpFilePaths.cmake )
 
-# Add the http_parser submodule to HTTP client library. It is out of the
-# unit-test tree and therefor needs it's own binary directory.
-add_subdirectory( ${MODULE_ROOT_DIR}/source/third-party ./build-http-parser )
-
-# Check if the http_parser source directory exists.
-if( NOT EXISTS ${HTTP_INCLUDE_THIRD_PARTY_DIRS}/http_parser.c )
-    # Attempt to clone http_parser.
-    if( ${BUILD_CLONE_SUBMODULES} )
-        clone_http_parser()
-    else()
-        message( FATAL_ERROR "The required submodule http_parser does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
-    endif()
-endif()
-
-# ============================ Test configurations =============================
-
-# Define a CMock resource path.
-set( CMOCK_DIR ${MODULE_ROOT_DIR}/test/unit-test/CMock CACHE INTERNAL "CMock library source directory." )
-
-# Include CMock build configuration.
-include( cmock_build.cmake )
-
-# Check if the CMock source directory exists, and if not present, clone the submodule
-# if BUILD_CLONE_SUBMODULES configuration is enabled.
-if( NOT EXISTS ${CMOCK_DIR}/src )
-    # Attempt to clone CMock.
-    if( ${BUILD_CLONE_SUBMODULES} )
-        clone_cmock()
-    else()
-        message( FATAL_ERROR "The required submodule CMock does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
-    endif()
-endif()
-
-# Add unit test and coverage configuration.
-
-# Use CTest utility for managing test runs. This has to be added BEFORE
-# defining test targets with add_test()
-enable_testing()
-
-# Add build targets for CMock and Unit, required for unit testing.
-add_cmock_targets()
-
-# Add function to enable CMock based tests and coverage.
-include( ${MODULE_ROOT_DIR}/tools/cmock/create_test.cmake )
-
-# Include build configuration for unit tests.
-include( unit_test_build.cmake )
-
-# Add a target for running coverage on tests.
-add_custom_target( coverage
-    COMMAND ${CMAKE_COMMAND} -DCMOCK_DIR=${CMOCK_DIR}
-    -P ${MODULE_ROOT_DIR}/tools/cmock/coverage.cmake
-    DEPENDS cmock unity core_http_utest core_http_send_utest
-    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-)
+# ====================  Define your project name (edit) ========================
+set(project_name "core_http")
+
+# =====================  Create your mock here  (edit)  ========================
+
+# list the files to mock here
+list(APPEND mock_list
+            ${HTTP_INCLUDE_THIRD_PARTY_DIRS}/http_parser.h
```

Comment:
If the variable {HTTP_INCLUDE_THIRD_PARTY_DIRS}, as the name suggests, had more than one directory, would it work?

Reply:
No it doesn't appear so. I will unplural that.